### ISSUE TITLE
Different number of potential dots in 1xso

### ISSUE THREAD

**Issue by russell-taylor** [ReliaSolve/cctbx_project#185]
There are a different number of potential dots in 1xso for Probe2 vs. probe when we count dots and show the summary tables.  This also impacts the score per square Angstrom line.

**Comment by russell-taylor** [comment]
There are 616738 in Probe2 and 616328 in Probe (slightly more in Probe2).

**Comment by russell-taylor** [comment]
There is replacement of dots with Carbonyl dots in the original code, but we should be using the correct radii for the Carbonyl atoms; perhaps that point set is using a different density?  Probably not, because there are Carbonyl atoms in 1bti which gets the correct answer.

**Comment by russell-taylor** [comment]
Dummy Hydrogens as source atoms are being excluded in both cases.

**Comment by russell-taylor** [comment]
(wrong) The code is not checking for ionic radius of ions.  Actually, this is being filled in within the Helpers.py routine.

**Comment by russell-taylor** [comment]
(done) When drawing the surfaces of 1xso in old and new, we get a number of extra dots in Probe (https://github.com/reliasolve/cctbx_project/issues/186), which is the opposite of what we see in the current case where there are more dots in Probe2.

**Comment by russell-taylor** [comment]
This count comes from self._count_skin_dots() which calls self._dotScorer.count_surface_dots() in Probe2 and enumDotSkin() which calls countSkin() in Probe.

**Comment by russell-taylor** [comment]
Is this an issue with occupancy in terms of which atoms are marked as bonded?  If anything this should make the Probe2 count smaller because we'd have fewer dots because more would be occluded.

**Comment by russell-taylor** [comment]
(done) Print the number of dots per atom in both and figure how whodunnit.

Lots of smaller differences rather than large differences on a couple of atoms.  Looks like the differences may be mostly around the ions:

```
$ diff old.txt new.txt
437c437
< XXX ATOM A ASP 81 OD1 count 211
---
> XXX ATOM A ASP 81 OD1 count 216
451c451
< XXX ATOM A CU 1 CU count 53
---
> XXX ATOM A CU 1 CU count 100
865c865
< XXX ATOM A HIS 118 NE2 count 142
---
> XXX ATOM A HIS 118 NE2 count 160
916c916
< XXX ATOM A HIS 44 ND1 count 91
---
> XXX ATOM A HIS 44 ND1 count 160
934c934
< XXX ATOM A HIS 46 NE2 count 140
---
> XXX ATOM A HIS 46 NE2 count 161
948c948
< XXX ATOM A HIS 61 HE2 count 33
---
> XXX ATOM A HIS 61 HE2 count 71
951c951
< XXX ATOM A HIS 61 NE2 count 87
---
> XXX ATOM A HIS 61 NE2 count 105
2341c2341
< XXX ATOM A ZN 152 ZN count 56
---
> XXX ATOM A ZN 152 ZN count 60
2778c2778
< XXX ATOM B ASP 81 OD1 count 214
---
> XXX ATOM B ASP 81 OD1 count 219
2792c2792
< XXX ATOM B CU 152 CU count 50
---
> XXX ATOM B CU 152 CU count 100
3206c3206
< XXX ATOM B HIS 118 NE2 count 141
---
> XXX ATOM B HIS 118 NE2 count 159
3257c3257
< XXX ATOM B HIS 44 ND1 count 94
---
> XXX ATOM B HIS 44 ND1 count 161
3275c3275
< XXX ATOM B HIS 46 NE2 count 154
---
> XXX ATOM B HIS 46 NE2 count 157
3289c3289
< XXX ATOM B HIS 61 HE2 count 43
---
> XXX ATOM B HIS 61 HE2 count 73
3292c3292
< XXX ATOM B HIS 61 NE2 count 90
---
> XXX ATOM B HIS 61 NE2 count 106
4024c4024
< XXX ATOM B MET 56 HB2 count 112
---
> XXX ATOM B MET 56 HB2 count 109
4688c4688,4695
< XXX ATOM B ZN 153 ZN count 52
---
> XXX ATOM B ZN 153 ZN count 56
```

**Comment by russell-taylor** [comment]
Not sure why we're seeing different counts here.  The ions are not being completely ignored; they do have some dots on them in both versions. (The only atoms marked to be ignored in probe.c are those whose occupancy or B values are out of range.)

Some of the contacts are with Hydrogens, most are with heavy atoms.  Maybe we're using a different radius for ions in the two cases?  We do get the "Using ionic radius" printed for CU and ZN.  The METAL_PROP, which is set for Zn and Cu, is not used anywhere in the original code other than to print info.  However, it does cause POSITIVE_PROP to be set, which is used... but not in the code that checks for potential dots.

The radius of some atoms is set to zero, but this is atoms that are ignored due to occupancy and B values.  The other place setting radius seems to be getRadius(), which will pull it from the atomprops.c AtomTblIndex structure, which matches the radii used by Probe2.

**Comment by russell-taylor** [comment]
The neighbor calculations do use a probe radius of 0 for Probe when calling findTouchingAtoms().  They also have marked the ionic atoms as bonded, which would reduce the number of dots placed on these atoms.  The ionic-bond calculation is done first in Probe2 so those should be included?  Many of the residues and atoms with ionic bonds are the ones that have very different counts.  Maybe it is allowing dots where there are clashes and one of the atoms is inside another.   There may be more of these in regions around the ions than elsewhere.

(nope) The CU atoms have 0 neighbors in Probe2 but 5 in Probe.  Probe looks for nearby touching atoms, not bonded atoms.  When we switch to nearby checks, this makes the CU atoms have the correct counts, but makes differences in a lot of other counts (we may not have the right radius); the counts are lower by from 1 to many for Probe2 when we use proximity. (nope) We're probably ignoring alternates.  Nope; checking for compatible conformations along with ignored atoms doesn't fix it.  We do have more neighbors for the CU in Probe2 (but they are ignored due to distance?).  Removing dummy hydrogens from those that are potential neighbors did not help.

(done) The markBonds() call in probe has different parameters for checking bonded atoms, a distance count of 1 and a max of 1.  But the basic problem seems to be that we're not including the ionically-bonded atoms, since some must be bonded to CU...  The problem was that we were recalculating the bonded atoms after doing a selection.  We are properly filtering irrelevant atoms so don't need to use a subset and can just use the same list.

(done) There was a less-than comparison on the distance in Probe2 but a less-than-or-equal in Probe.  This made no difference in the behavior.

**Comment by russell-taylor** [comment]
Fixed in 206a10305e3572ec99eb9cf513fc40af37a19912
